test/integration/serverTest.js

```diff
@@ -976,4 +1009,85 @@ describe('Brave Sensor server', () => {
       })
     })
   })
+
+  describe('POST request heartbeat events with mock INS Firmware State Machine', () => {
+    beforeEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+
+      const client = await clientFactory(db)
+      await db.createLocation(
+        testLocation1Id,
+        testLocation1PhoneNumber,
+        MOVEMENT_THRESHOLD,
+        STILLNESS_TIMER,
+        DURATION_TIMER,
+        1000,
+        INITIAL_TIMER,
+        ['+15005550006'],
+        '+15005550006',
+        ['+15005550006'],
+        1000,
+        'locationName',
+        radar_coreID,
+        radar_coreID,
+        'Innosent',
+        'alertApiKey',
+        true,
+        true,
+        null,
+        firstLowBatteryAlert,
+        client.id,
+      )
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+    })
+
+    afterEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+      sandbox.restore()
+      helpers.log('\n')
+    })
+
+    it('should return 200 for a heartbeat request with no headers', async () => {
+      const response = await chai.request(server).post('/api/heartbeat').send({})
+      expect(response).to.have.status(200)
+    })
+
+    it('should return 200 for a valid heartbeat request', async () => {
+      const request = {
+        coreid: radar_coreID,
+        data: `{"doorMissedMsg": 0, "doorLowBatt": true, "doorLastHeartbeat": 1000, "resetReason": "NONE", "states":[]}`,
+      }
+      const response = await chai.request(server).post('/api/heartbeat').send(request)
+      expect(response).to.have.status(200)
+    })
```

Comment:
Should also include tests for whether `client.xadd` from `redis.js` was called or not in these cases.